packages/cli-kit/assets/cli-ruby/lib/shopify_cli/theme/repl/api.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+module ShopifyCLI
+  module Theme
+    class Repl
+      class Api
+        attr_reader :ctx, :repl
+
+        def initialize(ctx, repl)
+          @ctx = ctx
+          @repl = repl
+        end
+
+        def request(liquid_snippet)
+          Net::HTTP.start(uri.host, 443, use_ssl: true) do |http|
+            req = Net::HTTP::Post.new(uri)
+
+            req.initialize_http_header(headers)
+            req.set_form_data(form_data(liquid_snippet))
+            res = http.request(req)
+
+            debug(res)
+
+            res
+          end
+        end
+
+        private
+
+        def debug(response)
+          return response unless debug?
+
+          ctx.debug(<<~DEBUG)
+            URI: #{uri}
+            ---
+            HTTP status: #{response.code}
+            ---
+            Response body:
+            #{response.body}
+          DEBUG
+
+          response
+        end
+
+        def debug?
+          @is_debug ||= ctx.debug?
+        end
+
+        def form_data(liquid_snippet)
+          template = ["", "", liquid_snippet, "", liquid_template].join("\n")
+
+          {
+            "replace_templates[sections/announcement-bar.liquid]" => template,
```

Comment:
Could there be themes that don't have `sections/announcement-bar.liquid`?

Reply:
Yes! I that's fixed in the new commit :)